cdap-unit-test/src/test/java/co/cask/cdap/admin/AdminApp.java

```diff
@@ -0,0 +1,423 @@
+import co.cask.cdap.api.spark.SparkContext;
+import co.cask.cdap.api.worker.AbstractWorker;
+import co.cask.cdap.api.workflow.AbstractWorkflow;
+import co.cask.cdap.api.workflow.AbstractWorkflowAction;
+import co.cask.cdap.internal.guava.reflect.TypeToken;
+import com.google.common.base.Throwables;
+import com.google.common.collect.Iterables;
+import com.google.gson.Gson;
+import org.apache.hadoop.io.LongWritable;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.mapreduce.Job;
+import org.apache.hadoop.mapreduce.Mapper;
+import org.apache.hadoop.mapreduce.Reducer;
+import org.apache.spark.api.java.JavaPairRDD;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.api.java.function.FlatMapFunction;
+import org.apache.spark.api.java.function.Function2;
+import org.apache.spark.api.java.function.PairFunction;
+import org.junit.Assert;
+import scala.Tuple2;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+
+public class AdminApp extends AbstractApplication {
+
+  public static final String FLOW_NAME = "AdminFlow";
+  public static final String MAPREDUCE_NAME = "AdminMapReduce";
+  public static final String SPARK_NAME = "AdminSpark";
+  public static final String SPARK_SCALA_NAME = "AdminScalaSpark";
+  public static final String SERVICE_NAME = "AdminService";
+  public static final String WORKER_NAME = "AdminWorker";
+  public static final String WORKFLOW_NAME = "AdminWorkflow";
+
+  @Override
+  public void configure() {
+    addStream("events");
+    addFlow(new AdminFlow());
+    addMapReduce(new AdminMapReduce());
+    addSpark(new AdminSpark());
+    addSpark(new AdminScalaSpark());
+    addWorker(new AdminWorker());
+    addWorkflow(new AdminWorkflow());
+    addService(new AbstractService() {
+      @Override
+      protected void configure() {
+        setName(SERVICE_NAME);
+        addHandler(new DatasetAdminHandler());
+      }
+    });
+  }
+
+  public static class DatasetAdminHandler extends AbstractHttpServiceHandler {
+
+    private static final Gson GSON = new Gson();
+
+    @GET
+    @Path("exists/{dataset}")
+    public void exists(HttpServiceRequest request, HttpServiceResponder responder,
+                       @PathParam("dataset") String dataset) throws DatasetManagementException {
+      Admin admin = getContext().getAdmin();
+      responder.sendString(Boolean.toString(admin.datasetExists(dataset)));
+    }
+
+    @GET
+    @Path("type/{dataset}")
+    public void type(HttpServiceRequest request, HttpServiceResponder responder,
+                     @PathParam("dataset") String dataset) throws DatasetManagementException {
+      Admin admin = getContext().getAdmin();
+      String type = admin.getDatasetType(dataset);
+      if (type == null) {
+        responder.sendStatus(404);
```

Comment:
Although, that makes the app depend on netty. Since CDAP does not expose constants for HTTP codes, I'd rather not use that.